packaging/technical-addon/Splunk_TA_otel/windows_x86_64/bin/Splunk_TA_otel_utils.ps1

```diff
@@ -1,63 +1,85 @@
+		otelLogWrite "INFO GrandParent process id: $grandParentId"		
+	}
+	catch {
+		# Use Write-Error directly here to get detailed information.
+		Write-Error "ERROR getParentAndGrandParent: $_"
+	}
+
+	return $parentId, $grandParentId
 }
 
-function isOtelProcessRunning($processName)
+function getOtelProcessPid($processName, $parentId)
 {
-	$parent = (Get-WmiObject win32_process | ? processid -eq $PID).parentprocessid
-	$Global:parentPid = $parent
-	otelLogWrite "INFO Parent process id: $parent"
-	$child = (Get-WmiObject win32_process | ? parentprocessid -eq $parent | ? processname -eq $processName).processid
-	$Global:otelPid = $child
-	otelLogWrite "INFO Collector process id: $child"
-	$otelProcess = Get-Process -Id $child
+	otelLogWrite "INFO Getting PID for $processName with parent PID $parentId ..."
+	$procPid = (Get-WmiObject Win32_Process | Where-Object ParentProcessId -eq $parentId | Where-Object ProcessName -eq $processName).ProcessId
+	otelLogWrite "INFO Collector process id: $procPid"
+	$otelProcess = Get-Process -Id $procPid -ErrorAction Ignore
 	if($otelProcess)
 	{
-		return $true
+		return $procPid
 	}
 	else
 	{
-		return $false
+		return -1
 	}
 }
 
-function waitForExit($parent) 
+function forceStopProcess($processId)
 {
-	Wait-Process -Id $parent
-	otelLogWrite "INFO Parent process exited"
+	otelLogWrite "INFO Stopping $processId if stil running"
+	# Use a pipeline to avoid an error message if the process already finished
+	Get-WmiObject Win32_Process | Where-Object ProcessId -eq $processId | ForEach-Object { taskkill.exe /f /pid $_.ProcessId }
 }
 
-function forceStopOtelProcess($processId)
-{
-	Stop-Process -Id $processId -Force
-	Start-Sleep -Seconds 1
-}
-
-function CheckOtelProcessStop($processId)
-{
-	$processInfo = Get-WmiObject win32_process | ? processid -eq $processId
-	if($processInfo) {
-		otelLogWrite "ERROR Otel agent didn't stop"
+otelLogWrite "INFO Splunk_TA_otelutils.ps1 started"
+try {
+	otelLogWrite "INFO sleeping ..."
+	start-sleep -s 3
+	otelLogWrite "INFO getting parent and grand parent PIDs"
+	$parentId, $grandParentId = getParentAndGrandParent $PID
+	otelLogWrite "INFO got parent $parentId and grand parent $grandParentId PIDs"
+	$otelPid = getOtelProcessPid $otelProcessName $parentId
+	if ($otelPid -eq -1) {
+		otelLogWrite "ERROR Otel agent not running"
 	} else {
-		otelLogWrite "INFO Otel agent stopped"
-	}
+		otelLogWrite "INFO Otel agent running"
+		$parentJob      = Start-Job { Wait-Process -Id $using:parentId } -Name "cmd job"
+		$grandParentJob = Start-Job { Wait-Process -Id $using:grandParentId } -Name "splunkd job"
+		$otelJob        = Start-Job { Wait-Process -Id $using:otelPid } -Name "Otel agent job"
+		otelLogWrite "INFO waiting on termination of any one of splunkd, cmd, and Otel agent"
+		$finishedJob = Wait-Job -Any -Job $parentJob,$grandParentJob,$otelJob
+		otelLogWrite "INFO at least one of the jobs finished:`n`t'$($finishedJob.Name)': finished with $($finishedJob.JobStateInfo)"
```

Comment:
Maybe include in this log message that the collector process is stopped as a result of this. (Please feel free to ignore if it'll be obvious in the logs, especially if the `waiting on termination` and `at least one of the jobs...` logs are going to be next to each other in the log file.)